src/StreetNameRegistry/Municipality/MunicipalityStreetName.cs

```diff
@@ -178,6 +184,26 @@ public void CorrectNames(Names names, Action<Names, PersistentLocalId> guardStre
             Apply(new StreetNameNamesWereCorrected(_municipalityId, PersistentLocalId, correctedNames));
         }
 
+        public void CorrectRetirement(Action guardUniqueActiveStreetNameNames)
+        {
+            if (IsRemoved)
```

Comment:
This IsRemoved check can go. This is now checked in Municipality when we do

`var streetName = StreetNames.GetNotRemovedByPersistentLocalId(persistentLocalId);`